_shared/project/tox.ini

```
@@ -55,26 +59,31 @@ deps =
     lint,tests,functests: factory-boy
     lint,tests,functests: h-matchers
     lint,template: cookiecutter
-{%- endif %}
-    {{- include("tox/deps", indent=4) }}
+{% endif %}
+    {% if include_exists("tox/deps") %}
+        {{- include("tox/deps", indent=4) -}}
+    {% endif %}
```

```suggestion
    {% if include_exists("tox/deps") %}
        {{ include("tox/deps", indent=4) }}
    {% endif %}
```

Do we need the whitespace stripping bits here?

---

Unfortunately we do need them. This:

```jinja2
Before
{% if include_exists("foo") %}
    {{ include("foo", indent=4) }}
{% endif %}
After
```

outputs this:

```
Before
        included_line_1
    included_line_2
    included_line_3

After
```

The leading `{{-` on the `include()` expression is needed to dedent the three lines from the `tox/deps` file. Alternatively we could not indent the `{{ include("tox/deps", indent=4) }}` line within the `{% if %}`. The trailing `-}}` is to get rid of the unwanted blank line after the `tox/deps` contents.

This is normal Jinja2 behaviour. For example if you were rendering a macro instead of calling our `include()` function it'd be the same:

```jinja2
{% macro foo() %}
macro_line_1
macro_line_2
macro_line_3
{% endmacro -%}

Before
{% if true %}
    {{- foo() -}}
{% endif %}
After
```